File: 1.1/src/c/cz/mg/vulkan/PFN_vkCmdBindVertexBuffers.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCmdBindVertexBuffers_callNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jint firstBinding, jint bindingCount, jlong pBuffers, jlong pOffsets){
    (void)env;
    (void)jc;
    PFN_vkCmdBindVertexBuffers f = (PFN_vkCmdBindVertexBuffers)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        (uint32_t)firstBinding,
        (uint32_t)bindingCount,
        ((VkBuffer*)jniLongToPointer(pBuffers)),
        ((VkDeviceSize*)jniLongToPointer(pOffsets))
    );
}

File: 1.1/src/cz/mg/vulkan/PFNvkCmdBindVertexBuffers.java
package cz.mg.vulkan;

public class PFNvkCmdBindVertexBuffers extends VkFunctionPointer {
    public PFNvkCmdBindVertexBuffers() {
    }

    protected PFNvkCmdBindVertexBuffers(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCmdBindVertexBuffers(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCmdBindVertexBuffers(long value) {
        setValue(value);
    }

    public PFNvkCmdBindVertexBuffers(VkInstance instance) {
        super(instance, new VkString("vkCmdBindVertexBuffers"));
    }

    public void call(VkCommandBuffer commandBuffer, int firstBinding, int bindingCount, VkBuffer pBuffers, VkDeviceSize pOffsets){
        callNative(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), firstBinding, bindingCount, pBuffers != null ? pBuffers.getVkAddress() : VkPointer.NULL, pOffsets != null ? pOffsets.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long commandBuffer, int firstBinding, int bindingCount, long pBuffers, long pOffsets);

}
